core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1813,7 +1813,7 @@ class Partition(val topicPartition: TopicPartition,
   private def submitAlterPartition(proposedIsrState: PendingPartitionChange): CompletableFuture[LeaderAndIsr] = {
     debug(s"Submitting ISR state change $proposedIsrState")
     val future = alterIsrManager.submit(
-      new TopicIdPartition(topicId.getOrElse(Uuid.ZERO_UUID), topicPartition),
+      new org.apache.kafka.server.common.TopicIdPartition(topicId.getOrElse(throw new IllegalStateException("Topic id not set")), topicPartition.partition),
```

Comment:
It's not safe to use `Uuid.ZERO_UUID` with the changes in this PR and hence we now throw an exception is the topic is is not known. @jolshan Is this OK?

Reply:
I wonder whether we should make the topic id mandatory (not an Option) in the Partition now. It is always set in KRaft anyway.

Reply:
@dajac I agree with you, but I thought that should be done in a separate PR as I think it will require some refactoring given the way various things interact right now. And perhaps it can be done in trunk only (no backport to 4.0 required).

Reply:
> I think this makes sense. I'm not sure exactly though what the error handling is here when an error is thrown. It seems like the futures here are not always checked. (Also wondering if we want to include more information about the partition in the error message)

My assumption is that the topic id is actually always set when this method is called, so the error thrown is just a way to make sure we are not missing something. Once we refactor things, this will be more obvious, but it would be nice to avoid blocking 4.0 on that.

Reply:
> if we want to include more information about the partition in the error message

Good idea - I will add the `topicPartition` to the error message.

Reply:
> so the error thrown is just a way to make sure we are not missing something. 

Yeah, that makes sense. The main thing I was wanting to check is that the error doesn't get swallowed somewhere. 👍